hack/update-codegen.sh

```diff
@@ -43,3 +43,6 @@ then
   fi
   mv sigs.k8s.io/kueue/client-go ${KUEUE_ROOT}/client-go
 fi
+
+# We need to clean up the go.mod file since code-generator adds temporary library to the go.mod file.
+"${GO_CMD}" mod tidy
```

Comment:
should we add it to the tools package instead?

Reply:
Ah, right.

Reply:
Uhm, tools don't seem work well.

```shell
go: finding module for package golang.org/x/crypto
sigs.k8s.io/kueue/hack/internal/tools imports
        golang.org/x/crypto: module golang.org/x/crypto@latest found (v0.10.0), but does not contain package golang.org/x/crypto
```

Reply:
:thinking: 

no idea

/lgtm
/approve